csrc/multidevice/utils.cpp

```diff
@@ -149,21 +149,22 @@ int64_t getShardedLogicalAxis(
     const TensorView* tv,
     const ParallelType parallel_type) {
   std::unordered_map<ParallelType, IterDomain*> parallel_type_to_id =
-      mapDeviceParallelTypeToId(tv->getMaybeAllocationDomain());
-  IterDomain* alloc_id = getOrDefault(parallel_type_to_id, parallel_type);
-  if (alloc_id == nullptr) {
+      mapDeviceParallelTypeToId(tv->getLoopDomain());
+
+  IterDomain* loop_id = getOrDefault(parallel_type_to_id, parallel_type);
```

Comment:
Using `loop_domain` in favor of `allocation_domain` since we may not have the allocation domain set in certain tests (`CommunicationTest`). Ideally, in real fusions, this should not be an issue.

Reply:
This is producing some errors later on in scheduling when reshape is involved, due to the `merge` transforms. Will have to investigate further.